lib/traject/readers/marc_combining_reader.rb

```diff
@@ -1,8 +1,41 @@
-class Traject::MarcCombiningReader < Traject::MarcReader
+class CombiningEnumerable
+  def initialize(left, right)
+    @left = left
+    @right = right
+  end
+
+  def each(*args, &block)
+    return to_enum(:each, *args) unless block_given?
+    @left.each(*args, &block)
+    @right.each(*args, &block)
+  end
+end
+
+class Traject::MarcCombiningReader
+  include Enumerable
+
+  attr_reader :settings, :input_stream, :marc_reader
+
+  def initialize(input_stream, settings)
+    @marc_reader = Traject::MarcReader.new(input_stream, settings)
```

Comment:
Does this then not use the `traject-marc4j_reader`? Are there side effects of this wrt performance?

Reply:
Perhaps, but adding support for marc4j reader is blocked by https://github.com/traject/traject-marc4j_reader/pull/2.